code/qc_utils.py

```diff
@@ -628,9 +632,45 @@ def generate_drift_qc(
         depth_at_max_cumulative_drift = int(spatial_bins[max_cumulative_drift_index])
 
         ax_drift.plot(temporal_bins, displacement_arr + spatial_bins, color="red", alpha=0.5)
+
+    # Plot surface channel line
+    surface_channel_y_position = None
+    if processing is not None:
+        try:
+            data_processes = processing.processing_pipeline.data_processes
+            for data_process in data_processes:
+                params = data_process.parameters.model_dump()
+                outputs = data_process.outputs.model_dump()
+                if (
+                    data_process.name == "Ephys preprocessing"
+                    and params.get("recording_name") is not None
+                    and params.get("recording_name") == recording_name
+                ):
+                    channel_labels = np.array(outputs.get("channel_labels"))
+                    if channel_labels is not None:
+                        channel_ids_out = recording.channel_ids[channel_labels == 'out']
+
+                        if not channel_ids_out:
+                            logging.info(f"No out channels detected for {recording_name}")
+                        else:
+                            surface_channel_id = channel_ids_out[0]
+                            surface_channel_index = recording.channel_ids.tolist().index(surface_channel_id)
+                            surface_channel_y_position = y_locs[surface_channel_index]
+                            ax_drift.axhline(y=surface_channel_y_position, c='g')
+        except:
+            logging.info(f"Failed to load bad channel labels for {recording_name}")
+
+    if surface_channel_y_position is not None:
+        ax_drift.set_title(
+            f"Max displacement: {max_displacement} $\mu m$ (depth: {depth_at_max_displacement} ) $\\mu m$\n"
+            f"Max cumulative drift: {max_cumulative_drift} $\mu m$ (depth: {depth_at_max_cumulative_drift} ) $\\mu m$\n"
+            f"Surface channel location (green line) at {y_position} $\\mu m$\n"
+        )
+    else:
```

Comment:
@arjunsridhar12345 there is one caveat here. Motion is estimated from the recording after detection and removal of out and bad channels, so the drift map will be somehow bound to the channels within the brain.

I think that what we could do to make the plot clearer is the following: 
- set ylim to min/max of ypos, so that we cover the probe y span even if drift map and motion is limited
- we could color the region from surface_channel_y_position and max(ypos) with a patch

@arjunsridhar12345 @jsiegle what do you guys think?

Reply:
After talking to @arjunsridhar12345 we thought that it might be better to add the surface channel in the raw data and psd metrics, since the drift map shouldn't have any spikes because out channels have been removed. @jsiegle is that ok?

Reply:
Yes that's fine!